Sources/SegmentFirebase/FirebaseDestination.swift

```diff
@@ -92,9 +100,16 @@ public class FirebaseDestination: DestinationPlugin {
         let name = formatFirebaseEventNames(event.event)
         var parameters: [String: Any]? = nil
         if let properties = event.properties?.dictionaryValue {
-            parameters = returnMappedFirebaseParameters(properties)
+            parameters = returnMappedFirebaseParameters(properties, for: FirebaseDestination.mappedKeys)
         }
-        
+
+        if let campaign = event.context?.dictionaryValue?["campaign"] as? [String: Any] {
+            let campaignParameters = returnMappedFirebaseParameters(campaign, for: FirebaseDestination.campaignMappedKeys)
+            let parameters = (parameters ?? [:]).merging(campaignParameters) { (current, _) in current }
+            FirebaseAnalytics.Analytics.logEvent(FirebaseAnalytics.AnalyticsEventCampaignDetails, parameters: parameters)
+            analytics?.log(message: "Firebase logEventWithName \(name) parameters \(String(describing: parameters))")
```

Comment:
Hmm I'm a little wary about this. I just looked at the docs for this event and I'm not sure we should be firing this on every single event? But rather just the first time we receive it? We need to confirm this with Jordan and the Segment team